tasks/DroneFlightControlTask.cpp

```diff
@@ -112,20 +106,19 @@ void DroneFlightControlTask::applyTransition(
      * is needed because there is no guarantee that a single try is enough, and the
      * feedback that this was successfull might be late.
      */
-    while (now < deadline && !canTakeControl(control_device))
+    while (now < deadline)
     {
         auto timeout = (deadline - now).toSeconds();
         mAuthorityStatus = mVehicle->obtainCtrlAuthority(timeout);
         ACK::getErrorCodeMessage(mAuthorityStatus, __func__);
         control_device = mVehicle->subscribe->getValue<Telemetry::TOPIC_CONTROL_DEVICE>();
+        if (!canTakeControl(control_device))
```

Comment:
This shouldn't be negated. We return true if we took control (i.e. `canTakeControl(control_device)` returns true), and return false if couldnt take control by the deadline.

Reply:
Now that the while logic is different, let's keep that negative.